### ISSUE TITLE
To much Fisting and just standing around

### ISSUE THREAD

**Issue by HooTmAnTW** [LordGolias/Brytenwalda#26]
For me the combat is broke, to many only fist bandits and village people and bad weapon equiping... for all units. Then most slingers and archers just stand around and do not switch to close combat, too.
And guards and speermen are still archers, why?, I put them always to Inf.

**Comment by LordGolias** [comment]
I agree. Unfortunately there are too many things to change in this regard.

One thing is the mission_template triggers. The other thing are items. I'm working on an overall change of items as follows:

Divide items in terms of type and level. For example, in shields, I was already able to separate all shields in terms of {faction, level, weight}. This way, in the `module_troops.py` we do not add to the troop inventory a specific shield, but a list of shields, e.g. `shields['saxon'][2]['heavy']` is the list of all saxon shields level 2 (from 0 to 2), and heavy (options: light, normal, heavy; heavy means more hit-points, less speed, more armor, more difficult).

The other trick is that shields properties are no longer defined on each item, but given by its type and level. For example, shield armor will be given by:

```
shield_type_to_int = {'light': 0, 'normal': 1, 'heavy': 2}
def shield_armor(lvl, shield_type):
    return 3 * lvl + shield_type_to_int[shield_type]
```

This way I'm decoupling game balance (stats of items) from game visual (which shield mesh), as the list of specific shields can be modified without modifying their stats and which troop carries it, and shield stats can be modified to groups of well defined shields. This also avoids the problem of having very shitty shields which cost allot, as the cost of the shield now scales with its stats:

```
def shield_cost(lvl, shield_type):
    return 2**lvl * (100 + 100 * shield_type_to_int[shield_type])
```
We can discuss the details of the equations, as I didn't though much about then, but the point stands: it is much easier to tweak game balance.

This will be done for all items, but it takes an hilarious amount of time as there is a thousand of items and troops, which is why I'm not commiting so much now to github.

**Comment by HooTmAnTW** [comment]
ok, thx for the answer that is much to change! For now is it possible? like gdwitt sad to give them more blunt or other weapons so they actually have some. And let the archers fight in close combat also with weapons? Makeing the Ai more agressive with close combat movement, also for Infantry they love to hold position and use jevallins then they just stand around for some time.. even they have no javelins anymore, they also start to fist fight with shields in the other hand. Best would be to just switch this behavior back to a older better working point, until there is a better new version, option?

**Comment by gdwitt** [comment]
I think the problem hoot is seeing is that they fight with fists even when given several weapons. Is the solution to give each troop more weapons in module_troops? I'm not sure and will ask at the forge.
I like the idea of having definitions for weapons and having a standardized table based on values. I even started a spreadsheet in SEptember and asked Magnar to do a version based on his work in Vae Victus. But I'd rather see a logical AI, good battles and realistic campaign/town systems first. 
However, that issue is labor intensive and may not be the biggest problem to fix right now.

**Comment by gdwitt** [comment]
"And guards and speermen are still archers, why?, I put them always to Inf."
There is a script in module_scripts or mission templates that autoassigns them to categories based on the number of missiles. This causes them to default stand up front instead of hiding in shield wall, which is good.  I am not sure what happens if you assign them ahead of battle.

**Comment by gdwitt** [comment]
I recalled that several scripts "force" throwing weapons on designated troops. There are also multiple scripts that attempt to assign them to weapon types given a formation. The new toggle script for blunts in mission_templates might be interfering. However, all of these are old hat and have been in the script for a while. I did alter the formation weapons script a little over the summer to get the troops to use spears. I will test it without the toggle script and with weapon assignments in module_troops. I am still not sure which troop types are resorting to fists. I was hoping that the new blunt toggle script would force the troops to pick a weapon and not use fists if they couldn't decide.

**Comment by gdwitt** [comment]
I added more melee items to all the bandit troop-types. It seems to fix the problem for the bandits.
My mercenary slinger and shepherd parties which I didn't fix are still fighting with fists.

Here is what Lumos thinks at [](https://forums.taleworlds.com/index.php/topic,6575.22185.html)
Are orders interfering? A forced command to use blunt weapons will force soldiers without blunt weapons to switch to unarmed.
Giving the same weapon several times doesn't "fix" the problem, but it increases the chance the troop will spawn with that weapon. There was a long discussion about this at some point several years ago, might want to look for that. IIRC, it culminated in the "guarantee all weapons" script, which does fix the problem and should be easily findable somewhere around here. (I do think there was some performance cost associated with using it, but that's life.)

**Comment by HooTmAnTW** [comment]
THX for letting me know! I had some performance probs with the new version at large battles with smart AI thing to ON, it was almost unplayable, so i switched it OFF and performance was getting way better but still not that good like it was befor. 
Also the blunt orders may did give the units only fists? For version 24 patch A it was way better (less fisting), well some units just do not have weapons or arrows like the Mercenary Archers.
